### PR TITLE
Add multi-thread ver olaptablesink for LOAD

### PR DESCRIPTION
**Pull request by vagetablechicken**
Ref https://github.com/apache/incubator-doris/issues/2780#issuecomment-588156273
So we can use 
```
LOAD LABEL ...
        PROPERTIES 
        ( 
        "buffer_num"="5", 
        "mem_limit_per_buf"="5368709120", 
        "size_limit_per_buf"="62914560"
        );
```
to use multi-thread version olaptablesink,  to accelerate the time-consuming load.
If we don't set buffer_num or set it to 0, use the origin version(single-thread).

- [x] add configuration references

- [x] add doc

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/load/loadv2/LoadJob.java [hunk 1]
```diff
@@ -17,6 +17,20 @@
 
 package org.apache.doris.load.loadv2;
 
+import com.google.common.base.Joiner;
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/load/loadv2/LoadJob.java]: Plz reorganize your imports order:

org.apache.doris
com
org
java
javax

File: fe/src/main/java/org/apache/doris/planner/OlapTableSink.java [hunk 2]
```diff
@@ -88,12 +92,29 @@ public OlapTableSink(OlapTable dstTable, TupleDescriptor tupleDescriptor, List<L
...
     public void init(TUniqueId loadId, long txnId, long dbId, long loadChannelTimeoutS) throws AnalysisException {
         TOlapTableSink tSink = new TOlapTableSink();
         tSink.setLoad_id(loadId);
         tSink.setTxn_id(txnId);
         tSink.setDb_id(dbId);
         tSink.setLoad_channel_timeout_s(loadChannelTimeoutS);
+
+        if (bufferNum > 0) {
+            tSink.setBuffer_num(bufferNum);
+            if (memLimitPerBuf > 0)
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/planner/OlapTableSink.java]: warp with `{}`, even if there is only one line after `if`

File: gensrc/thrift/DataSinks.thrift [hunk 3]
```diff
@@ -105,6 +105,9 @@ struct TOlapTableSink {
     12: required Descriptors.TOlapTableLocationParam location
     13: required Descriptors.TPaloNodesInfo nodes_info
     14: optional i64 load_channel_timeout_s // the timeout of load channels in second
+    15: optional i32 buffer_num
```
Comment (morningman) [inline gensrc/thrift/DataSinks.thrift]: Add some comment for these fields

File: be/src/exec/tablet_sink.h [hunk 4]
```diff
@@ -141,26 +191,75 @@ class IndexChannel {
...
 
-// The counter of add_batch rpc of a single node
-struct AddBatchCounter {
-    // total execution time of a add_batch rpc
-    int64_t add_batch_execution_time_ns = 0;
-    // lock waiting time in a add_batch rpc
-    int64_t add_batch_wait_lock_time_ns = 0;
-    // number of add_batch call
-    int64_t add_batch_num = 0;
+class RowBuffer {
```
Comment (morningman) [inline be/src/exec/tablet_sink.h]: Add some comments for this class

**Review by morningman — CHANGES_REQUESTED** [review]

**Comment by vagetablechicken** [comment]
After talking the matter over, we decided against this multi-thread version. I will submit a new design, just use one sender thread send batches asynchronously.
Thanks for the reviews!

**Pull request closed without merging**